### ISSUE TITLE
`PLR6301` (`no-self-use`) gives false positive for overriding method that does not call super()

### ISSUE THREAD

**Issue by jakob-keller** [astral-sh/ruff#9799]
I ❤️  ruff!

While playing with the current set of preview rules, I stumbled upon this edge case: 

```python
# test.py


class Foo:
    def __init__(self) -> None:
        self.value: int = 1

    def get_value(self) -> int:
        return self.value


class Bar(Foo):
    def get_value(self) -> int:
        return 2
```

```
$ ruff --version
ruff 0.2.0

$ ruff check --isolated --preview --select PLR test.py
test.py:13:9: PLR6301 Method `get_value` could be a function, class method, or static method
   |
12 | class Bar(Foo):
13 |     def get_value(self) -> int:
   |         ^^^^^^^^^ PLR6301
14 |         return 2
   |

Found 1 error.
```

Admittedly, the example is a bit construed. Yet I believe this is valid code and should not trigger an error.

By the way: PyCharm correctly handles this case. On the other hand, ruff is (currently) not a type checker and might be unable to do anything about it.

**Comment by charliermarsh** [comment]
Thank you so much for the kind words!

I'm torn on what to do here (in the short-term, while we don't support multi-file analysis). We could...

1. Detect this when the parent class is defined in the same file (as in the example above).
2. Limit this rule to only flag for classes without any parent classes.
3. Nothing?

By the way, one workaround here: if you decorate the method with `@typing_extensions.override`, we'll avoid flagging it.

**Comment by jakob-keller** [comment]
> if you decorate the method with `@typing_extensions.override`, we'll avoid flagging it.

Great suggestion! Our codebase is still on Python 3.11 and doesn't use the override decorator yet. But I will probably use the backport available from `typing_extensions`. That's the most elegant solution I can think of.

I am closing this issue in favor of option 3 (won't fix) since:

> 1. Detect this when the parent class is defined in the same file (as in the example above).

Is rarely the case in substantial codebases.

> 2. Limit this rule to only flag for classes without any parent classes.

Introduces too many false negatives IMHO.
